### ISSUE TITLE
Training and testing result of kpconv on toronto3d do not make sense

### ISSUE THREAD

**Issue by chdjianfeng** [isl-org/Open3D-ML#256]
@sanskar107 
Thank you for your great work. I am very interested in your project, but I encountered the following problems when training and testing kpconv model on toronto3d. Looking forward to your reply. Thank you very much!
1.Testing problem
When I used the predefined scripts "ml3d/configs/kpconv_toronto3d.yml" and weight file "https://storage.googleapis.com/open3d-releases/model-zoo/kpconv_toronto3d_202012221551utc.pth" to test the kpconv model on toronto3d, I got a overall accuracy of 82.02% and MIOU of 46.4. That is lower than your score MIOU 65.6.
2.Training problem
Then I used the predefined scripts "ml3d/configs/kpconv_toronto3d.yml" to train  the kpconv model on toronto3d from scratch. The results do not make sense.
![图片](https://user-images.githubusercontent.com/46151512/114639733-223ec800-9d01-11eb-9732-f82e800d5537.png)
![图片](https://user-images.githubusercontent.com/46151512/114639818-54e8c080-9d01-11eb-8869-bf4d927f137e.png)

**Comment by sanskar107** [comment]
@chdjianfeng there was a small bug in IoU calculation which is fixed in the `dev` branch. If you take mean of ious of each individual pointcloud in the test set, you get mIoU around 46. But what we want is mIoU obtained from the combined confusion matrix which comprises of whole test set. This will be fixed in the next release.
Toronto3D is a tricky dataset during training, some specific preprocessing is required to reproduce the results, I'll try to train and get back to you.

**Comment by chdjianfeng** [comment]
@sanskar107 OK! Thank you very much!

**Comment by Hao-HUST** [comment]
> @sanskar107
> Thank you for your great work. I am very interested in your project, but I encountered the following problems when training and testing kpconv model on toronto3d. Looking forward to your reply. Thank you very much!
> 1.Testing problem
> When I used the predefined scripts "ml3d/configs/kpconv_toronto3d.yml" and weight file "https://storage.googleapis.com/open3d-releases/model-zoo/kpconv_toronto3d_202012221551utc.pth" to test the kpconv model on toronto3d, I got a overall accuracy of 82.02% and MIOU of 46.4. That is lower than your score MIOU 65.6.
> 2.Training problem
> Then I used the predefined scripts "ml3d/configs/kpconv_toronto3d.yml" to train the kpconv model on toronto3d from scratch. The results do not make sense.
> ![图片](https://user-images.githubusercontent.com/46151512/114639733-223ec800-9d01-11eb-9732-f82e800d5537.png)
> ![图片](https://user-images.githubusercontent.com/46151512/114639818-54e8c080-9d01-11eb-8869-bf4d927f137e.png)

Did you use the tf or torch version testing kpconv model on toronto3d? It seems that the torch version pipeline.test() cannot output the accuracy and mean IoU.

**Comment by chrise96** [comment]
@sanskar107 RandLa-Net on Toronto-3D also gets weird results. This is a visualization of the val set, with the pretrained model... It Looks like it predicts one class. Is the fix already in the master branch or is it in the dev branch?

<img width="1025" alt="Schermafbeelding 2021-07-27 om 14 16 32" src="https://user-images.githubusercontent.com/22835281/127152860-a20f1189-dc93-4721-aa75-b499786b1b91.png">

**Comment by sanskar107** [comment]
Closing due to inactivity. Feel free to reopen in case you face similar problems again.
